storm-client/src/jvm/org/apache/storm/daemon/GrouperFactory.java

```diff
@@ -137,7 +137,7 @@ public void prepare(WorkerTopologyContext context, GlobalStreamId stream, List<I
     public static class FieldsGrouper implements CustomStreamGrouping {
 
         private Fields outFields;
-        private List<Integer> targetTasks;
+        private ArrayList<List<Integer> > targetTasks;
```

Comment:
Unless you are using some specific method of the implementation its always better to write code against the interfaces.